build/testdata/bundles/wordpress/porter.yaml

```diff
@@ -26,6 +26,10 @@ parameters:
 - name: wordpress-password
   type: string
   sensitive: true
+- name: mysql-password
```

Comment:
I don't understand why you introduced this parameter and then later down flipped `externaldatabase.password` to go from being retrieved from an output to being supplied by a parameter. This really changes the intent of dependencies and what the bundle was demonstrating.

Reply:
I did this as an exercise to explore how a dependency parameter might be overridden from a bundle that uses that dependency, but I can imagine that I didn't do it quite right and/or we don't wish to add for this particular bundle.

I'll remove my override of the dep chart param (`mysql-password`) from params/outputs and just use the `wordpress-password` for testing.

Though, in general, would it not be a valid use case to override arbitrary bundle dependency params via the bundle that utilizes these dependencies?  For this particular example, consider the case where the user wants to ensure the mysql/external db password is a certain value whilst installing the wordpress bundle that uses the mysql/external db bundle.

Reply:
Oh, right, I see that any param overrides would go in the `dependencies` section under the appropriate dependency bundle 😅 .  So, perhaps, for the stated use case above, if a dependency param needed to be dynamically set/overridden (on `porter install`, etc.), we'd need to allow for this in the cli... (read: future feature, if valid)

Reply:
We had our hands tied with dependency parameters originally (because they had to be merged ultimately into a single manifest/invocation image). So the way it originally worked doesn't matter too much, I'm in the process of deleting that code as I type. But yes, it was designed so that you would default a parameter on a dependency in the dependencies section like so:

```
dependencies:
- name: mysql
  parameters:
    database-name: wordpress
```

I'd like to be able to also pass through a parameter defined on the current bundle (something that wasn't possible with the original design). Something like this:

```
dependencies:
- name: debug
  parameters:
    debug: {{ porter.parameters.debug }}
```

The user will need a way to be able to control any parameter (from the main bundle or dependency). Anything set from the cli wins. Though there's a bit of design/thought to go into dealing with how to handle specifying params for deps when parameters can now have the same name between a deps and the main bundle, it's not as simple as it was before.

I think this is all out of scope of the current PR (?), it falls into the dependencies rewrite I'm doing now.